Funcky.Test/Extensions/EnumerableExtensions/PowerSetTest.cs

```diff
@@ -0,0 +1,44 @@
+using System.Collections.Generic;
+using System.Linq;
+using Funcky.Extensions;
+using Funcky.Test.TestUtils;
+using Xunit;
+
+namespace Funcky.Test.Extensions.EnumerableExtensions
+{
+    public sealed class PowerSetTest
+    {
+        [Fact]
+        public void APowerSetIsEnumeratedLazily()
+        {
+            var doNotEnumerate = new FailOnEnumerateSequence<object>();
+
+            _ = doNotEnumerate.PowerSet();
+        }
+
+        [Fact]
+        public void ThePowerSetOfTheEmptySetIsSetOfTheEmptySet()
+        {
+            var powerSet = Enumerable.Empty<string>().PowerSet();
+            Assert.Empty(powerSet.First());
+        }
+
+        [Fact]
+        public void ThePowerSetIsTheSetOfAlLSubSets()
+        {
+            var set = new List<string> { "Alpha", "Beta", "Gamma" };
+            var powerSet = set.PowerSet();
+
+            Assert.Collection(
```

Comment:
Does `PowerSet` guarantee anything about the ordering of the subsets?

Reply:
No, that is an implementation Detail, but the test was easier to write this way :D